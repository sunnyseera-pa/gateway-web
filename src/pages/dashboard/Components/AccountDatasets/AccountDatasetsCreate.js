import React, { useEffect } from 'react';
import _ from 'lodash';
import { Alert, Button, Col, Row } from 'react-bootstrap';
import SVGIcon from '../../../../images/SVGIcon';
import serviceDatasetOnboarding from '../../../../services/dataset-onboarding/dataset-onboarding';
import googleAnalytics from '../../../../tracking';

const AccountDatasetsCreate = props => {
	const {
		publisherID,
		team,
		alert: { message },
	} = props;

	const dataPostDatasetOnboarding = serviceDatasetOnboarding.usePostDatasetOnboarding({ publisherID }, null, {
		enabled: false,
	});

	const createNewDataset = e => {
		e.preventDefault();

		dataPostDatasetOnboarding.refetch();
	};

	useEffect(() => {
		if (dataPostDatasetOnboarding.data) {
			const {
				data: {
					data: {
						data: { id },
					},
				},
			} = dataPostDatasetOnboarding;

			if (!_.isUndefined(id)) window.location.href = `/dataset-onboarding/${id}`;
		}
	}, [dataPostDatasetOnboarding.data]);

	return (
		<>
			{message && (
				<Alert variant={'success'} className='col-sm-12 main-alert'>
					<SVGIcon name='check' width={18} height={18} fill={'#2C8267'} /> {message}
				</Alert>
			)}
			<div className='accountHeader'>
				<Row>
					<Col sm={12} md={8}>
						<div>
							<span className='black-20'>Datasets</span>
						</div>
						<div>
							<span className='gray700-13 '>
								{publisherID !== 'admin'
									? 'View, add, edit, archive and check the status of your datasets.'
									: 'Approve or reject pending datasets'}
							</span>
						</div>
					</Col>
					<Col sm={12} md={4} style={{ textAlign: 'right' }}>
						{team !== 'admin' && (
							<Button
								variant='primary'
								className='addButton'
								onClick={
									(() => googleAnalytics.recordEvent('Datasets', 'Add a new dataset', 'Datasets dashboard button clicked'),
									createNewDataset)
								}>
								+ Add a new dataset
							</Button>
						)}
					</Col>
				</Row>
			</div>
		</>
	);
};

export default AccountDatasetsCreate;
