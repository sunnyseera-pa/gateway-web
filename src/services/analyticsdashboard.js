import { useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest } from '../utils/requests';

const getTotalUsers = options => {
	return getRequest(`${apiURL}/analyticsdashboard/totalusers`, options);
};

const getUsersPerMonth = options => {
	return getRequest(`${apiURL}/analyticsdashboard/userspermonth`, options);
};

const useGetTotalUsers = (requestOptions, queryOptions = { queryKey: 'getTotalUsers' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getTotalUsers(requestOptions),
	});
};

const useGetUsersPerMonth = (requestOptions, queryOptions = { queryKey: 'getUsersPerMonth' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getUsersPerMonth(requestOptions),
	});
};

export default {
	getTotalUsers,
	getUsersPerMonth,
	useGetTotalUsers,
	useGetUsersPerMonth,
};
