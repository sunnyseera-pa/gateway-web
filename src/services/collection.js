import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest } from '../utils/requests';

export const getCollectionRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/${_id}`, options);
};

export const getCollectionRelatedObjectsRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/relatedobjects/${_id}`, options);
};

export const postCollectionCounterUpdateRequest = (data, options) => {
	return postRequest(`${apiURL}/collectioncounter/update`, data, options);
};

const useGetCollectionRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRequest(queryKey[1], requestOptions),
	});
};

const useGetCollectionRelatedObjectsRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRelatedObjectsRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRelatedObjectsRequest(queryKey[1], requestOptions),
	});
};

const usePostCollectionCounterUpdateRequest = (requestOptions, mutateOptions = { queryKey: 'postCollectionCounterUpdateRequest' }) => {
	return useMutation(data => postCollectionCounterUpdateRequest(data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCollectionRequest,
	getCollectionRelatedObjectsRequest,
	postCollectionCounterUpdateRequest,
	useGetCollectionRequest,
	useGetCollectionRelatedObjectsRequest,
	usePostCollectionCounterUpdateRequest,
};
