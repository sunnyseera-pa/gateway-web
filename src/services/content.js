import { useQuery } from 'react-query';
import { getRequest } from '../utils/requests';
import { UATCMSURL } from './content.constants';

const baseURL = require('../pages/commonComponents/BaseURL');
const cmsURL = baseURL.getCMSURL();
const env = baseURL.getURLEnv();
const local = 'local';

export const getNon5SafesModalContentRequest = (options) => {
    const url = env === local ? UATCMSURL : cmsURL;
	return getRequest(`${url}/Non5SafesModalContent`, options);
};

const useGetNon5SafesModalContentRequest = (requestOptions, queryOptions = { queryKey: 'getNon5SafesModalContentRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getNon5SafesModalContentRequest(requestOptions),
	});
};

export default {
    useGetNon5SafesModalContentRequest
}
