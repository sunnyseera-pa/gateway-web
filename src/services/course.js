import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest, putRequest } from '../utils/requests';

const getCourse = (_id, options) => {
	return getRequest(`${apiURL}/course/${_id}`, options);
};

const getEdit = (_id, options) => {
	return getRequest(`${apiURL}/course/edit/${_id}`, options);
};

const postCourse = (data, options) => {
	return postRequest(`${apiURL}/course`, data, options);
};

const putCourse = (_id, data, options) => {
	return putRequest(`${apiURL}/course/${_id}`, data, options);
};

const useGetCourse = (requestOptions, queryOptions = { queryKey: 'getCourse' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getCourse(_id, requestOptions),
	});
};

const useGetEdit = (requestOptions, queryOptions = { queryKey: 'getEdit' }) => {
	return useQuery({
		...queryOptions,
		queryFn: _id => getEdit(_id, requestOptions),
	});
};

const usePostCourse = (requestOptions, mutateOptions = { queryKey: 'postCourse' }) => {
	return useMutation(data => postCourse(data, requestOptions), {
		mutateOptions,
	});
};

const usePutCourse = (requestOptions, mutateOptions = { queryKey: 'putCourse' }) => {
	return useMutation((_id, data) => putCourse(_id, data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCourse,
	getEdit,
	postCourse,
	putCourse,
	useGetCourse,
	useGetEdit,
	usePostCourse,
	usePutCourse,
};
