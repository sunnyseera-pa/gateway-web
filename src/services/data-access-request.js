import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { deleteRequest, getRequest, patchRequest, postRequest, putRequest } from '../utils/requests';

const getDataAccessRequests = options => {
	return getRequest(`${apiURL}/data-access-request`, options);
};

const getDataAccessRequest = (_id, options) => {
	return getRequest(`${apiURL}/data-access-request/${_id}`, options);
};

const postDataAccessRequest = (_id, data, options) => {
	return postRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const putDataAccessRequest = (_id, data, options) => {
	return putRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const patchDataAccessRequest = (_id, data, options) => {
	return patchRequest(`${apiURL}/data-access-request/${_id}`, data, options);
};

const deleteDataAccessRequest = (_id, options) => {
	return deleteRequest(`${apiURL}/data-access-request/${_id}`, options);
};

const useGetDataAccessRequests = (requestOptions, queryOptions = { queryKey: 'getDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getDataAccessRequests(requestOptions),
	});
};

const useGetDataAccessRequest = (requestOptions, queryOptions = { queryKey: 'getDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: id => getDataAccessRequest(id, requestOptions),
	});
};

const usePostDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'postDataAccessRequest' }) => {
	return useMutation((id, data) => postDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const usePutDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'putDataAccessRequest' }) => {
	return useMutation((id, data) => putDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const usePatchDataAccessRequest = (requestOptions, mutateOptions = { queryKey: 'patchDataAccessRequest' }) => {
	return useMutation((id, data) => patchDataAccessRequest(id, data, requestOptions), {
		mutateOptions,
	});
};

const useDeleteDataAccessRequest = (requestOptions, queryOptions = { queryKey: 'deleteDataAccessRequest' }) => {
	return useQuery({
		...queryOptions,
		queryFn: id => deleteDataAccessRequest(id, requestOptions),
	});
};

export default {
	getDataAccessRequests,
	getDataAccessRequest,
	postDataAccessRequest,
	putDataAccessRequest,
	patchDataAccessRequest,
	deleteDataAccessRequest,
	useGetDataAccessRequests,
	useGetDataAccessRequests,
	useGetDataAccessRequest,
	usePostDataAccessRequest,
	usePutDataAccessRequest,
	usePatchDataAccessRequest,
	useDeleteDataAccessRequest,
};
