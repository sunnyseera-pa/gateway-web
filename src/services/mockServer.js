import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mswDatasets from './datasets/mockMsw';
import mswDatasetOnboarding from './dataset-onboarding/mockMsw';
import mswPostDatasetActivityLog from './activitylog/mockMsw';
import mswSearch from './search/mockMsw';
import mswGetLocations from './locations/mockMsw';
import mswGetContributors from './contributors/mockMsw';

import translations from '../../public/locales/en-GB/translation.json';

const mswGetEnTranslations = rest.get(`http://localhost/locales/en/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetEnGbTranslations = rest.get(`http://localhost/locales/en-GB/translation.json`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.json(translations));
});

const mswGetIcon = rest.get(`/images/Application_approved.svg`, (req, res, ctx) => {
	return res(ctx.status(200), ctx.text('<svg />'));
});

const handlers = [
	...mswDatasets,
	...mswDatasetOnboarding,
	...mswPostDatasetActivityLog,
	// ...mswSearch,
	...mswGetLocations,
	...mswGetContributors,
	mswGetEnTranslations,
	mswGetEnGbTranslations,
	mswGetIcon,
];

export const server = setupServer(...handlers);
