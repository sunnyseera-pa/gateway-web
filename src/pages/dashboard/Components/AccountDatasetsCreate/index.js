import AccountDatasetsCreate from './AccountDatasetsCreate';

export default AccountDatasetsCreate;
