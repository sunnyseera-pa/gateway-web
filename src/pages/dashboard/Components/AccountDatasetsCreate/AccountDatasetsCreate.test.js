import { render, createEvent, waitFor } from '@testing-library/react';
import React from 'react';
import AccountDatasetsCreate from '.';
import { server } from '../../../../services/mockServer';

let wrapper;

const props = { publisherID: 'admin', team: '1234', alert: { message: 'Sample message' }, isFederated: false, isLoading: false };

describe('Given the AccountDatasetsCreate component', () => {
    beforeAll(() => {
        server.listen();
    });

    afterEach(() => {
        server.resetHandlers();
    });

    afterAll(() => {
        server.close();
    });

    describe('When it is rendered', () => {
        beforeAll(async () => {
            wrapper = render(<AccountDatasetsCreate {...props} />, {
                wrapper: Providers,
            });
        });

        it('Then matches the previous snapshot', () => {
            expect(wrapper.container).toMatchSnapshot();
        });

        describe('And "Add new dataset" is clicked', () => {
            it('Then calls preventDefault', async () => {
                const button = wrapper.queryByText(/Add a new dataset/i);
                const buttonEvent = createEvent.click(button, { cancelable: true });

                fireEvent.click(button, buttonEvent);

                expect(wrapper.container).toMatchSnapshot();
            });
        });

        describe('And the user is federated', () => {
            it('Then hides create new datatset', async () => {
                wrapper.rerender(<AccountDatasetsCreate {...props} isFederated />);

                expect(wrapper.queryByText(/Add a new dataset/i)).toBeNull();
            });
        });

        describe('And data is loading', () => {
            it('Then hides create new datatset', async () => {
                wrapper.rerender(<AccountDatasetsCreate {...props} isLoading />);

                expect(wrapper.queryByText(/Add a new dataset/i)).toBeNull();
            });
        });
    });
});
