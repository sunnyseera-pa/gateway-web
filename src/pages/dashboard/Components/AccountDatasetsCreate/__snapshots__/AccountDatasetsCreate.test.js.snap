// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AccountDatasetsCreate component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="fade col-sm-12 main-alert alert alert-success show"
    role="alert"
  >
    <svg
      class=""
      data-testid="check"
      height="18"
      viewBox="0 0 24 22"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
      xmlns:xlink="http://www.w3.org/1999/xlink"
    >
      <path
        d="M9.988 17c-.265 0-.52-.105-.707-.292l-3-3c-.38-.393-.374-1.016.012-1.402.386-.386 1.01-.391 1.402-.012l2.2 2.2 6.303-8.107c.339-.437.967-.516 1.404-.176.436.339.515.967.176 1.404l-7 9c-.176.227-.441.368-.728.386h-.062z"
        fill="#2C8267"
      />
    </svg>
     
    Sample message
  </div>
  <div
    class="accountHeader"
  >
    <div
      class="row"
    >
      <div
        class="col-md-8 col-sm-12"
      >
        <div>
          <span
            class="black-20"
          >
            datasets
          </span>
        </div>
        <div>
          <span
            class="gray700-13 "
          >
            dataset.create.description.user
          </span>
        </div>
      </div>
      <div
        class="col-md-4 col-sm-12"
        style="text-align: right;"
      >
        <button
          class="addButton btn btn-primary"
          type="button"
        >
          + 
          dataset.create.action
        </button>
      </div>
    </div>
  </div>
</div>
`;
